### PR TITLE
rename PersistenceId.apply to PersistenceId.ofUniqueId, #27924

### PR DESCRIPTION
**Pull request by patriknw**
* to make the entityTypeHint, entityId the more prominent choice, and
  avoid that only the entityId is used as the uniqueId

Refs #27924

### CODE REVIEW

File: akka-persistence-typed/src/main/scala/akka/persistence/typed/PersistenceId.scala [hunk 1]
```diff
@@ -122,18 +122,28 @@ object PersistenceId {
...
+    new PersistenceId(id)
+
 }
 
 /**
  * Unique identifier in the backend data store (journal and snapshot store) of the
  * persistent actor.
  */
-final case class PersistenceId(id: String) {
+final class PersistenceId private (val id: String) {
```
Comment (patriknw) [inline akka-persistence-typed/src/main/scala/akka/persistence/typed/PersistenceId.scala]: Changed to class to get rid of the apply, but also good to make it easier to evolve if we need some other change in the future.

I don't think there is a need for unapply.

File: akka-cluster-typed/src/test/scala/akka/cluster/typed/ClusterSingletonPersistenceSpec.scala [hunk 2]
```diff
@@ -33,7 +33,7 @@ object ClusterSingletonPersistenceSpec {
 
   val persistentActor: Behavior[Command] =
     EventSourcedBehavior[Command, String, String](
-      persistenceId = PersistenceId("TheSingleton"),
+      persistenceId = PersistenceId.ofUniqueId("TheSingleton"),
```
Comment (johanandren) [inline akka-cluster-typed/src/test/scala/akka/cluster/typed/ClusterSingletonPersistenceSpec.scala]: That is one mouthful, could we just keep the `.of` and call the parameter `uniqueId` instead?

Comment (patriknw) [reply akka-cluster-typed/src/test/scala/akka/cluster/typed/ClusterSingletonPersistenceSpec.scala]: `.of(` is already taken for the ordinary Java factory methods, so would have same problem as stated in the motivation of this change.

`ofUniqueId` is a power user api so will not be used by many

Comment (johanandren) [reply akka-cluster-typed/src/test/scala/akka/cluster/typed/ClusterSingletonPersistenceSpec.scala]: Ah, missed that aspect. Then it's fine.

**Review by johanandren — APPROVED** [review]

**Comment by patriknw** [comment]
pls review @renatocaval

**Review by octonato — APPROVED** [review]
LGTM, thanks @patriknw.

While going through the docs, I spotted typo:

On `persistence.md`, we have the following sentence:

> To create a unique `PersistenceId` the `entityId` is can be prefixed
with a stable name of the entity type, ...

I think it should read:

> To create a unique `PersistenceId` the `entityId` should be prefixed
with a stable name of the entity type, ...

**Comment by patriknw** [comment]
thanks, I have fixed the typo

**Pull request merged**